concordium-node/src/grpc2.rs

```diff
@@ -484,6 +520,65 @@ pub mod types {
         }
     }
 
+    impl TryFrom<AccountTransactionV1Signatures>
+        for concordium_base::common::types::TransactionSignaturesV1
+    {
+        type Error = tonic::Status;
+        fn try_from(value: AccountTransactionV1Signatures) -> Result<Self, Self::Error> {
+            fn convert_account_transaction_signature(
+                signature: AccountTransactionSignature,
+            ) -> Result<
+                BTreeMap<
+                    concordium_base::common::types::CredentialIndex,
+                    BTreeMap<
+                        concordium_base::common::types::KeyIndex,
+                        concordium_base::common::types::Signature,
+                    >,
+                >,
+                tonic::Status,
+            > {
+                signature
+                    .signatures
+                    .into_iter()
+                    .map(|(ci, m)| {
+                        let ci = u8::try_from(ci).map_err(|_| {
+                            tonic::Status::invalid_argument("Invalid credential index.")
+                        })?;
+                        let cred_sigs = m
+                            .signatures
+                            .into_iter()
+                            .map(|(ki, sig)| {
+                                let ki = u8::try_from(ki).map_err(|_| {
+                                    tonic::Status::invalid_argument("Invalid key index.")
+                                })?;
+                                let sig = sig.try_into()?;
+                                Ok::<_, tonic::Status>((ki.into(), sig))
+                            })
+                            .collect::<Result<BTreeMap<concordium_base::common::types::KeyIndex, _>, _>>()?;
+                        Ok::<_, tonic::Status>((ci.into(), cred_sigs))
+                    })
+                    .collect::<Result<
+                        BTreeMap<
+                            concordium_base::common::types::CredentialIndex,
+                            BTreeMap<concordium_base::common::types::KeyIndex, concordium_base::common::types::Signature>,
+                        >,
+                        _,
+                    >>()
+            }
+            // TODO(drsk) Why is the sender_signatures an optional field?
+            let sender_signatures =
+                convert_account_transaction_signature(value.sender_signatures.require()?)?;
+            let sponsor_signatures = match value.sponsor_signatures {
+                Some(s) => Some(convert_account_transaction_signature(s)?),
+                None => None,
+            };
```

Comment:
Wait.. This just needs to be `TransactionSignature` type from base, right? Are you referencing an old version of https://github.com/Concordium/concordium-base/pull/781 where this is not the case?

Reply:
I think I understand now. I updated base and now changed this to `TransactionSignature`.